### PR TITLE
LongClassNameCheck

### LINKED ISSUE #81: Check for class name more than x symbols
Class names with more 13 symbols reveal that code is too complex.
We should prevent it, by rejecting build if such check fails.
check: https://www.yegor256.com/2015/01/12/compound-name-is-code-smell.html

### PR DESCRIPTION
**Pull request by h1alexbel**
@l3r8yJ take a look, please
closes #81

---

### Focus of this PR:
- Introduces a new rule for checking long class names
- Updates the PDD workflow to use a different action
- Adds test cases for the new rule
- Updates the `ValidateMojo` class to include the new rule
- Adds a new parameter for specifying the maximum length of class names

### Detailed summary:
- Added `LongClassNameCheck` rule to check for long class names
- Updated PDD workflow to use `volodya-lombrozo/pdd-action@master`
- Added test cases for `LongClassNameCheck`
- Updated `ValidateMojo` class to include `LongClassNameCheck` rule
- Added `maxClassNameLen` parameter to `ValidateMojo` for specifying maximum class name length

> The following files were skipped due to too many changes: `src/it/long-class-name/pom.xml`, `src/main/java/ru/l3r8y/rule/CompositeLongClassName.java`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by zoeself** [comment]
@h1alexbel thank you for your Pull Request. I'll assign someone to review it soon. 

 If this PR solves a ``todo`` from the code, please don't forget to remove it.

**Comment by zoeself** [comment]
@l3r8yJ please review this Pull Request. Deadline (when it should be merged or closed) is ``2023-12-31T14:00:20.904875``.

You should check if the requirements have been implemented (partially or in full), if there are unit tests covering the changes and if the CI build passes. Feel free to reject the PR or ask for changes if it's too big or not clear enough.

Estimation here is ``30 minutes``, that's how much you will be paid. You will be paid even if this PR gets rejected.

File: src/main/java/ru/l3r8y/ValidateMojo.java [hunk 1]
```diff
@@ -54,19 +55,35 @@ public final class ValidateMojo extends AbstractMojo {
...
 
+    /**
+     * Length of Class Name.
+     *
+     * @see ru.l3r8y.rule.LongClassNameCheck
+     * @checkstyle MemberNameCheck (6 lines).
+     */
+    @SuppressWarnings("PMD.ImmutableField")
+    @Parameter
+    private int okClassNameLength = 13;
```
Comment (l3r8yJ) [inline src/main/java/ru/l3r8y/ValidateMojo.java:73]: @h1alexbel i think this parameter can be configurable for each project, can you do it?

```xml
<maxClassNameLen>13<maxClassNameLen>
```

**Review by l3r8yJ — CHANGES_REQUESTED** [review]
@h1alexbel take a look at my comments below, please

**Review by l3r8yJ — APPROVED** [review]
@h1alexbel looks good to me!

**Comment by l3r8yJ** [comment]
@h1alexbel can you remove `pdd` GHA in this pr, looks like it's broken

**Comment by h1alexbel** [comment]
@l3r8yJ I will fix it

**Comment by h1alexbel** [comment]
@l3r8yJ take a look once again, please

**Comment by l3r8yJ** [comment]
@rultor merge

**Comment by zoeself** [comment]
@l3r8yJ thank you for resolving this ticket. I've just added it to your active invoice. You can always check all your invoices and more on the [Contributor Dashboard](https://self-xdsd.com).

**Pull request merged**